Use NDK 23 only for Windows users.

cortinico opened the pull request:
Summary:
Bumping the NDK to 23 to prevent build failures due to the NDK
using longer paths.

Changelog:
[Android] [Fixed] - Use NDK 23 only for Windows users.

Differential Revision: D35547459

RalissonMattias commented:
I can help in the tests, basically just change the file?

cortinico commented:
> @cortinico I can help in the tests, basically just change the file?

You can find simple instructions on how to test it here: https://gist.github.com/cortinico/6a297ac4c6a2a4633f58f1e4b375dd88

RalissonMattias commented:
> > @cortinico I can help in the tests, basically just change the file?
> 
> Thanks for that @RalissonMattias
> 
> You can find simple instructions on how to test it here: https://gist.github.com/cortinico/6a297ac4c6a2a4633f58f1e4b375dd88
> 
> Please publish the gradle scans once you're done with it.

I tested here and it didn't work
![Captura de tela 2022-04-11 133734](https://user-images.githubusercontent.com/92331757/162788511-8a4df9bd-b912-4deb-8d6f-6c483051eacd.png)

Ralisson-Mattias commented:
After applying the new patch, generate new error
![image](https://user-images.githubusercontent.com/68711690/162837936-1d284ba0-2635-4d05-a496-40b5697d5ea7.png)

cortinico commented:
> After applying the new patch, generate new error

Thanks for the feedback. I ended up finding a windows machine and finding a (hopefully) working solution. It works locally for me but I'd like someone else to give it another try. 

I've updated the .patch file in gist (here https://gist.github.com/cortinico/6a297ac4c6a2a4633f58f1e4b375dd88) with a new one (fix33611-v3.patch). Could you give it another try (make sure to delete the project and re-create it).

RalissonMattias commented:
Here still no success.

First git applyt throws errors:
![image](https://user-images.githubusercontent.com/92331757/162961135-0831f7c9-9756-4368-a20d-5dd2aad29cff.png)

After manually inserting only the `fix33611-v3.patch`:
![image](https://user-images.githubusercontent.com/92331757/162961947-3dc72121-2a57-4e16-b8d3-e2fe7afc9744.png)

After manually combining and entering patches v1 and v3:
![image](https://user-images.githubusercontent.com/92331757/162962553-bf2ebc51-7797-4234-a621-24af8962783c.png)

mikehardy commented:
@cortinico I gave it a go. 
-- ❌  First build failed https://gradle.com/s/qzgdhi6kbxlbc
-- ❌  Second build still failed, using v3 of patch from gist https://gradle.com/s/uly2zz74hljzo
  -- ironically, when I run `git apply fix33611-v3.patch` as downloaded via "raw" view mode + save file from browser, in powershell I get "Corrupt file at line 40" which seems to be some sort of unix vs dos patch format thing. Using `unix2dos` did not result in success but using `notepad fix33611-v3.patch` and just adding a newline at end of file + save + close resulted in patch that successfully applied 🤷 

```

  make: *** Deleting file 'C:\Users\micro\work\react-random\RN068\.cxx/local/armeabi-v7a/librrc_slider.a'
  C:/Users/micro/AppData/Local/Android/Sdk/ndk/21.4.7075529/build//../toolchains/llvm/prebuilt/windows-x86_64/bin/arm-linux-androideabi-ar: C:\Users\micro\work\react-random\RN068\.cxx/local/armeabi-v7a/objs-debug/rrc_slider/C_/Users/micro/work/react-random/RN068/node_modules/react-native/ReactAndroid/__/ReactCommon/react/renderer/components/slider/platform/android/react/renderer/components/slider/SliderMeasurementsManager.o: No such file or directory
  make: *** [C:/Users/micro/AppData/Local/Android/Sdk/ndk/21.4.7075529/build//../build/core/build-binary.mk:600: C:\Users\micro\work\react-random\RN068\.cxx/local/armeabi-v7a/librrc_slider.a] Error 1
  make: *** Waiting for unfinished jobs....
  make: *** Deleting file 'C:\Users\micro\work\react-random\RN068\.cxx/local/armeabi-v7a/librrc_progressbar.a'
  C:/Users/micro/AppData/Local/Android/Sdk/ndk/21.4.7075529/build//../toolchains/llvm/prebuilt/windows-x86_64/bin/arm-linux-androideabi-ar: C:\Users\micro\work\react-random\RN068\.cxx/local/armeabi-v7a/objs-debug/rrc_progressbar/C_/Users/micro/work/react-random/RN068/node_modules/react-native/ReactAndroid/__/ReactCommon/react/renderer/components/progressbar/android/react/renderer/components/progressbar/AndroidProgressBarShadowNode.o: No such file or directory
  make: *** [C:/Users/micro/AppData/Local/Android/Sdk/ndk/21.4.7075529/build//../build/core/build-binary.mk:600: C:\Users\micro\work\react-random\RN068\.cxx/local/armeabi-v7a/librrc_progressbar.a] Error 1
  make: *** Deleting file 'C:\Users\micro\work\react-random\RN068\.cxx/local/armeabi-v7a/librrc_switch.a'
  C:/Users/micro/AppData/Local/Android/Sdk/ndk/21.4.7075529/build//../toolchains/llvm/prebuilt/windows-x86_64/bin/arm-linux-androideabi-ar: C:\Users\micro\work\react-random\RN068\.cxx/local/armeabi-v7a/objs-debug/rrc_switch/C_/Users/micro/work/react-random/RN068/node_modules/react-native/ReactAndroid/__/ReactCommon/react/renderer/components/switch/androidswitch/react/renderer/components/androidswitch/AndroidSwitchShadowNode.o: No such file or directory
  make: *** [C:/Users/micro/AppData/Local/Android/Sdk/ndk/21.4.7075529/build//../build/core/build-binary.mk:600: C:\Users\micro\work\react-random\RN068\.cxx/local/armeabi-v7a/librrc_switch.a] Error 1
```

-- ✔️  Advised to move project from `C:\Users\micro\work\react-random\RN068` to root of C: and re-try, so, a recompile while it is in `C:\RN068` **worked** https://gradle.com/s/2yhpn6nkfxjro

cortinico commented:
> -- ✔️ Advised to move project from `C:\Users\micro\work\react-random\RN068` to root of C: and re-try, so, a recompile while it is in `C:\RN068` **worked** [gradle.com/s/2yhpn6nkfxjro](https://gradle.com/s/2yhpn6nkfxjro)

Awesome thanks for giving it a try. I'll close this PR as we'll merge this instead:
- https://github.com/facebook/react-native/pull/33582

Which contains the v3 patch I've added in the gist.

I hope that with RN 0.69, as we moved to CMake, those kind of problems won't appear anymore.

cortinico commented:
> I know you already closed PR but I wanted to report that following the steps of @mikehardy worked correctly here too. Good job, see you later

Thanks for your feedback @Ralisson-Mattias 🙏 Yup I've closed this one as we're going to merge the other PR: https://github.com/facebook/react-native/pull/33582
Thanks again for testing this on your windows machine.

mganandraj commented:
On my trials, ndk23 is able to handle long paths, but the build fails because of long commmand lines (>8192 characters) .. I've created another PR with the ndk bump too ..
